subsys/net/openthread/Kconfig

```diff
@@ -13,14 +13,12 @@ config OPENTHREAD_NRF_SECURITY
 	depends on !OPENTHREAD_MBEDTLS
 	select NORDIC_SECURITY_BACKEND
 	select OBERON_BACKEND
-	select MBEDTLS_VANILLA_BACKEND
```

Comment:
Nothing wrong, but what is the reasons behind this Kconfig file, and the one here:
https://github.com/nrfconnect/sdk-nrfxlib/blob/master/openthread/Kconfig

Relates to this comment:
https://github.com/nrfconnect/sdk-nrfxlib/pull/244#discussion_r480937765

Could you try to align those two PRs and in each Kconfig file (if both are required), then make a description in each file on what kind of configuration should go where.

Reply:
@piotrkoziar still waiting for an answer ?

Reply:
There are three OpenThread Kconfig files in three repos:
- zephyr - Primary file related to the OT sources in zephyr
- nrf - exists to extend/override some settings that upstream zephyr should not be aware of. Settings applicable for building from sources or common to both sources and libs
- nrfxlib - settings related to openthread precompiled libraries only.

Reply:
added following description to the file:
# The purpose of this file is to override and extend settings related to
# building Thread from sources. It extends the zephyrs openthread Kconfig file.
# Separate Kconfig exists in nrfxlib repository responsible solely for managing
# OpenThread precompiled libraries.